### DISCUSSION TITLE
برخی پیشنهادات بعد بررسی پنل جدید و آپدیت

### DISCUSSION THREAD

**Discussion by lvlrf** [hiddify/Hiddify-Manager#439] [Ideas ایدهها]
درود بر شما وقتتون بخیر باشه

بعد این آپدیت گسترده که عملا میشه گفت پنل آپگرید شد چقدر همه چی عالی شد، کار آمد و کار راه بنداز، دست مریزا، متشکریم

تو لیست زیر یه سری پیشنهادات مطرح می کنم که برای من مهم بود و بهشون احساس نیاز داشتم، البته این پیشنهادام بعد بررسی پنل های محتلف کافکا، اسدی، صنائی و ... هستش، من کد های نرسیدم بررسی کنم، و میدونم که اصل هیدیفای سادگی و بدون پیچیدگی برای کاربر و کار کردن با هر سطح دانشی باهاش بود.

1.
خیلی ترجیحم بر این هست که به صورت Docker نصب کنم و استفاده کنم، بابت اینکه بتونم بی دردسرتر کنارش سرویس دیگه اجرا کنم یا اینکه سرویس دیگم خراب کاری کرد روی این کانتینر خراب کاری نکنه. از طرفی یکی از روش های دورزدن میکروتیک هستش که پروتکل های خیلی زیادی رو ساپورت می کنه، برای من شبکه کار که میکروتیک هم بلدم خیلی خوبه که تو ورژن 7 میکروتیک میشه کانتینر روش نصب کرد؛ عملا من نیازی نیست دوتا هزینه بدم برای سرورم.
 دیدم تو صفحه داکر زدید که در حال تست و آزمایشی هستش، سر فرصت حتما بررسیش می کنم، اما متاسفانه لینکشو از خود هاب داکر ندیدم

2.
فکر میکنم پنل به صورت اتوماتیک خودش Certificate میگیره، و روش های مختلفی هست که میشه certificate گرفت، من خودم خیلی دوست دارم سیستم قابلیت شخصی سازی داشته باشه، (چون الآن پنل داره نیاز رو برطرف می کنه، پیشنهادام برای توسعه هستش) مثلا دوست داشتم یه سری تنظیمات رو از من موقع نصب می پرسید من میگفتم آره یا نه، یا اینکه مثلا یه نصب آسان داشت و یه نصب شخصی سازی شده تر، از طرفی میدونم همه چی توی قسمت تنظیمات میشه دید و ویرایش کرد.

از بین روش های SSL گرفتن من برای کلادفلر رو خیلی دوست دارم، به چند دلیل، مهم ترینش اینه که فیلترینگ معمولا دامنه و ساب دامنه هستش (اکثرا هم روی  A رکورد ساب دامنه میارن بالا ( ولی توی کلادفلر میشه روی کل دامنه و همه ساب دامنه ها گواهینامه SSL گرفت، عملا اگر کسی سرورش هم فیلتر شد، دیگه نیازی نداره کار خاصی بکنه چه بابت گواهینامه و ... فقط میتونه از سرورش یه SnapShot بگیره و با آی پی جدید یا Domian جدید بیاره بالا دامنه های رو تغییر بده و مشکلش رو برطرف کنه (دیدم که خیلی ها سوال پرسیده بودن اگه فیلتر شدیم چی کار کنیم راه کاری قطعی براشون نبود)، یه خوبی دیگه میشه 15 ساله گرفت، من تو پنل صنائی توی ترمینال با بایت acme تونستم با ایمیلی که باهاش توی کلادفلر ثبت نام کردم و کد API Zone ID دامنم مستقیم از کلاد فلر گواهیمو بگیرم و خیلی با این روش حال کردم.
اینم روش های احرازش بود
 "1.need Cloudflare account associated email"
 "2.need Cloudflare Global API Key"
 "3.your domain use Cloudflare as resolver"

از طرفی من یه سری ها رو دیدم رفت گواهینامه خریدن، یا یکی ممکنه دوست داشته باشه از Let's Encrypt یا ... گواهی بگیره، برای من چیزی که خیلی مهمه این گواهی های رایگان که مثلا 3ماهش تموم میشه به سوای اینکه تمدید خودکار داشته باشه، ایمیلمو داده باشم و بتونم مطلع بشه، اگه مثلا یه قسمت توی تنظیمات بود مربوط به SSL که روش گرفتش و اینکه مسیر گواهی ها خیلی خوب میشد، من خودم خیلی بیشتر لذت میبردم ازش.

3.
من خیلی دوست داشتم تو قسمت پروکسی ها بتونم خودم دستی تنظیم کنم که مثلا Trojan-GRPC-TLS از پورت 2096 استفاده کنه، اما متاسفانه حق انتخابی ندارم، میدونم این قضیه یه بحث امنیت داره، یک بحث خطای کاربر داره که ممکنه پورتی بزاره سرویس های دیگش رو مختل کنه یا اینکه روی CDN اون پورت باز نباشه، ولی دوست دارم این قضیه روی با مسئولیت خودم و انتخاب خودم باشه نه که همه روی پورت 80 یا 443 باشن (احساس می کنم پورت ها رو تفکیک کنم بازخورد بهتری هم میشه گرفت حداقل که روی  CDN کلاد فلر 6تا پورت داریم دوست دارم ازشون استفاده کنم)

4.
ازونجایی که معمولا دامنه ها فیلتر میشن (اکثرا هم ساب دامنه ها) خیلی دوست داشتم فرضا وقتی دامنه با CDN معرفی می کنه توی یه دونه تعریف دامنه بتونیم چند تا ساب دامنه بدم با این نیت که هر سری توی آپدیت سابسکریپشن کاربر یه دامین رندم بهش بده که کاربر همیشه به یه ساب وصل نشه، هم ریسک فیلتر شدن رو تغییر بده و دامنه هام فرضن سخت تر لو بره. امیدوارم تونسته باشم منظورم رو رسونده باشم، که توی فیلد اول دامنه بتونم چندتا دامنه استفاده کنم و با کاما بتونم جداشون کنم.

5.
فایل ترجمه
این رو سیاست گروه تعیین می کنه اما من توی اسکریپت های چند زبان میدیدم خودشون یه قسمت ترجمه داشتن که تگ اصلی داشتن و بر مبنای هر زبان یه فابل ترجمه داشتن، و بعضی هاشون به تفکیک پنل ادمین و پنل کاربر داشتن
نیاز اصلی من از این قضیه این هست که من برای استفاده از CDN به تفکیک اپراتور دامنه تعریف کردم واسه آی پی سالم و اسم کانفیگ هام خیلی طولانی شده،  دوست دارم اون اسم مستعار دامنه اول بیاد ادامشم خودم بتونم مخفف بدم واسه کانفیگ یا اینکه دیدم یه سری ها نگران بودن دوست ندارم مشتریم بدونه کانفیگ سالم چیه مثلا ندونن Vless-WS-TLS هست که مثلا دیر تر لو بره یا ... با این تفاسیر دستشون زیاد باز نیست.

6.
یه چیز دیگه که به نظر جای خالی حس میشه اینه که برای قسمت کاربران قسمت پکیج بستن هستش، مثلا توی پکیج میشه دسترسی به پروتکل ها رو تعیین کرد، زمان و اعتبار و ترافیک کاربر رو
دیدم چند نفر دوست داشتن خودشون تعیین کنن کودوم کاربر به چی دسترسی داشته باشه و کی به چی دسترسی نداشته باشه. این پکیج کمک میکنه واسه این مورد، از طرفی توی سیستم های اکانتیگ مرسوم هسستش که بسته ها رو میسازن سری بعد یوزر به پکیج اختصاص میدن، نهایت توی همون صفحه یه سری تغییرات میدن که کاربر از پکیج خارج میشه.
یه چیزی که واسه قسمت کاربران جای خالیش واقعا حس میشه، من دوست دارم مثلا اعتبار یه ماهه به کاربر میدم از اولین اتصالش باشه، ازونجایی که لاگ در حد آخرین اتصال هست یا ترفیک مصرفی میشه شرط گذاشت اولی مگ مصرفی کاربر شروع زمانش برای تاریخ انقضاش باشه.
و یه مورد دیگه اینکه ساختن دسته جمعی اکانت، متوجه شدم یوزر UUID داره و نام کاربری فقط توی پنل هست، و بک آپ خروجیش json هست، اگه بشه که csv کرد فکر میکنم خیلی ها بتونن استفاده کنن، و فرضا اگر کسی فیلد پکیج UUID یا ... کاربر رو وارد نکرد خود سیستم اتوماتیک یه چیز بهش بده.

7.
میدونم به کاربر اکانت میدیم ترجیحا سابسکریپشن دیگه لینک کاربریشو چک نمیکنه، و فقط تو اپ خودش میمونه، توی پنل های x-ray core دیگه مثل v2Board یا XMPlus که نگاه میکردم توی کانفیگ هایی که به کاربر میداد یه 2تا کانفیگ اولی که میداد الکی بود، عملا کانکشنی نبود؛ فقط اولی بهش اعتبار زمانیشو میگفت 3 روز مونده تاریخ فلان تموم میشه دومی هم ترافیک باقمونده از ترافیک کل ترافیکش رو بهش میگفت به نظرم واقعا جای این مورد تو پنل خالیه

8.
دیدم پنل قابلیت چند پنلی رو داره، اما مستنداتی برای نحوه استفاده ازش ندیدم، اما من دوست دارم، یه پنل کلی داشته باشم که تنظیمات رو توی اون بزنم که بقیه ازون تنظیمات رو ارث بری کنن(هدف rely اینه که از جای دیگه وصل شد، من منظورم پنل دیگه هست مثلا فرقشون توی لوکیشن سرور باشه)  و قسمت پکیج ها میتونه اینجا خیلی کمک کنه که کاربر به کودوم سرور ها دامنه ها و کانفیگ ها دسترسی داشته باشه.

9.
شخصا بابت نحوه ورود به پنل زیاد راحت نیستم، میدونم که این روش خوبیه هستش، اما فرضا من اگه لینک رو یه جا باز بزارم یا یکی کپی کنه برای خودش دسترسی ادمین داره، به سوالی لینک ورود غیر پورت و ادمین بعد آدرس که امنیت رو میبره بالا برای شناسایی پنل و... که خیلی ایده خوبی هست خیلی دوست داشتم توی قسمت تنظیمات بتونم تیک بزنم مثلا من یوزر پسوورد هم میخوام، ورود دو مرحله ای میخوام.

و چیزی که دوست دارم اینه که بات تلگرام برای مدیر داشته باشه که آخر شب بک آپ فول بگیره بفرسته اتوماتیک تایمی که براش تعریف شده و اینکه کسی لاگین کرد بهم اطلاع بده و از طرفی منابع مصرفی زیاد شد بهم اطلاع بده مثلا بالای 80% داشت میرفت اینا برای من صاحب سرور مهمه.

برای من قسمت rely روی سرور ایران کار نکرد، روی ipv6 رو هم همینطور اما بررسی می کنم، مشکلش رو پیدا کردم نتیجشو بازخورد میزارم شاید مشکل بقیه هم باشه حل بشه.

مرسی از صبوریتون
امیدوارم روزش برسه مجبور نباشیم برای استفاده از حق بدیهیمون اینترنت مجبور به تایم انرژی گذاشتن برای این جور چیزا نباشیم

**Comment by lvlrf** [comment]
10.
ای کاش تو قسمت تعریف دامنه تعیین میکردیم که روی این دامنه چه پروکسی هایی کار کنه اعم از تلگرام و CDN ها و بدون CDN ها

الآن من یه rely ساختم IPv6 و شنیدم که روی IPv6 فیلترینگی نداریم فعلنه حالا  الآن مشکلم اینه که من فقط میخوام پروکسی های با CDN کار کنه و با دامنه relyIPv6 فقط دایرکت ها کار کنه، اما الآن اگه اینکار رو بکنمDirect های دامنه های دیگمم باز میشه که نمیخوام

یا یدونه rely ایران آوردم بالا اون باید عملا دایرکت باشه و CDN زدن براش زیاد معنا نداره اما نمیخوام دایرکت پروکسی ها دیگه رو روشن کنم.


احساس می کنم توی فیلد آخر مربوط به دامنه که میگید روی لینک کاربرد این دامنه کانفیگ هایی ازدامنه های دیگه ای از نشون بده پروکسی هارو بزارید تیکشونو بزنیم این خیلی کاربردی تره

با پیشنهادم بابت تعریف پکیج هم میتونیم به تفکیک کاربرد تفکیک کنیم دسترسی به دامنه ها رو اینجوری فکر کنم به هم هدف آیتم قبلی که انجام میشه هم افرادی که مثل من مشکل دارن مشکلشون حل میشه.

**Comment by lvlrf** [comment]
11.
دیدم @iSegaro گفته که دامنه ها اخیرا سریع فیلتر میشن، و یکی از علت هاش میتونه حداقل ورژن TLS توی CloudFlare باشه، من جایی رو تو پنل ندیدم بتونم تنظیمش کنم، واسه پنل صنایی @MHSanaei میشه تنظیمش کرد
[عکس پنل](https://raw.githubusercontent.com/MHSanaei/3x-ui/main/media/1.png)
اینم اضافه بشه لطفا که خیلی واجب هستش

**Reply by iSegaro** [reply]
توی پنل نیست توی اکانت کلادفلرت روی دامین مورد نظرت باید فعلا کنی عکسش که گذاشتم
دامنهایی که ورژن TLS رو همون دیفالت 1 تنظیم کردن ساب دامنشون خیلی زود فیلتر میشه
دامنه من الان پابلیکه رو اینترنت چندین ماهه از اولم ورژن 1.3 TLS روش ست بوده حتی یکباره ساب دامنم مشکل نخورده
سیستم فیلترینگ ماشینی هست برای همین اگه قرار بود فیلتر بشم از همه زودتر باید میشدم ولی تا الان مشکلی نداشتم
